pkg/services/check/check.go

```diff
@@ -0,0 +1,120 @@
+package check
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+var ErrKubernetesNotFound = errors.New("no kubernetes version found")
+
+const (
+	FluxCompatibleMessage    = "Current flux version is compatible"
+	FluxNotCompatibleMessage = "Current flux version is not compatible"
+)
+
+// Pre runs pre-install checks
+func Pre(ctx context.Context, kubeClient kube.Kube, fluxClient flux.Flux, expectedFluxVersion string) (string, error) {
+	output := ""
+
+	k8sOutput, err := runKubernetesCheck(fluxClient)
+	if err != nil {
+		return "", err
+	}
+
+	output += k8sOutput + "\n"
+
+	currentFluxVersion, err := getCurrentFluxVersion(ctx, kubeClient)
+	if err != nil {
+		if errors.Is(err, kube.ErrNamespaceNotFound) {
+			output += "✔ Flux is not installed"
+			return output, nil
+		}
+
+		return "", fmt.Errorf("failed getting installed flux version: %w", err)
+	}
+
+	fluxOutput, err := validateFluxVersion(currentFluxVersion, expectedFluxVersion)
+	if err != nil {
+		return "", err
+	}
+
+	output += fluxOutput
```

Comment:
@josecordaz We need a newline character here. This is what I get:

```
jordan@UbuntuDesktop:~/Projects/weave-gitops$ ./bin/gitops check --pre
✔ Kubernetes 1.21.1 >=1.19.0-0
✗ Flux 0.21.0 !=~0.24.x
Current flux version is not compatiblejordan@UbuntuDesktop:~/Projects/gitopstest/weave-gitops$ 
```

What I would like is:

```
jordan@UbuntuDesktop:~/Projects/weave-gitops$ ./bin/gitops check --pre
✔ Kubernetes 1.21.1 >=1.19.0-0
✗ Flux 0.21.0 !=~0.24.x
Current flux version is not compatible
jordan@UbuntuDesktop:~/Projects/gitopstest/weave-gitops$ 
```